### ISSUE TITLE
Package proposal: bochs.vm

### ISSUE THREAD

**Issue by av-gantimurov** [mandiant/VM-Packages#897]
### Package Name

bochs

### Tool Name

Bochs x86 PC emulator

### Package type

SINGLE_EXE

### Is the tool a console application?

false

### Tool's version number

2.7

### Category

Debuggers

### Tool's authors

Donald Becker, Greg Alexander, Christophe Bothamy, Volker Ruppert, Bryce Denney, Stanislav Shwartsman

### Tool's description

Bochs is a portable x86 PC emulation software package that emulates enough of the x86 CPU, related AT hardware, and BIOS to run Windows, Linux, *BSD, Minix, and other OS's, all on your workstation. Used as debugger in IDA Pro (Local Bochs debugger)

### Download URL

https://sourceforge.net/projects/bochs/files/bochs/2.7/Bochs-win64-2.7.exe/download

### Download SHA256 Hash

966104a81b4d83599e07158f6dd35fe1

### Why is this tool a good addition?

After installing bochs may be used as Local Bosch Debugger in IDA Pro. IDA Pro has special plugin for Bochs support as debugger for x86 an x64 code.

**Comment by Ana06** [comment]
@av-gantimurov thanks for the proposal! :+1:  The provided hash is not an SHA256 hash, can you please update the issue description with the correct hash?

Can you please also provide a link to the IDA plugin in https://github.com/mandiant/VM-Packages/issues/719? :smiley:

**Comment by av-gantimurov** [comment]
> @av-gantimurov thanks for the proposal! 👍 The provided hash is not an SHA256 hash, can you please update the issue description with the correct hash?
> 
> Can you please also provide a link to the IDA plugin in #719? 😃

I wrote MD5 instead of SHA256. Fixed.
IDA Pro supports local bochs debugger without any extra plugins by default.
I added some links with information about using Bochs debugger in first message.

**Comment by Ana06** [comment]
> I wrote MD5 instead of SHA256. Fixed.

Our bot processes the package proposals and send a PR for it when we add the `send PR` label, but the information needs to be accurate/as expected. That's why we can't use the MD5 hash. Thanks for correcting it. As you can see, our bot has already send a PR and we will be able to get this in very soon 😉 

> IDA Pro supports local bochs debugger without any extra plugins by default. Worked with IDA 8.3 on Windows 10 without fault.
> I added some links with information about using Bochs debugger in first message.

We will add Bochs without without the plugin and close this issue. So I was proposing you suggest/document the Bosch plugin in [the existent issue about plugins additions](https://github.com/mandiant/VM-Packages/issues/719) so that we don't forget about it and we get it in as well. We don't have automation for plugins (yet). 😄

**Comment by Ana06** [comment]
@mandiant/flare-vm should we add bochs to the default flare-vm installation? I think it would be nice to have more debuggers available that can be used with IDA. 🤔 Opinions?

**Comment by Ana06** [comment]
Sadly the package type was not correct, the downloaded exe is not the tool itself but an installer and our bot does not support this package type (yet). Does someone want to send a PR for this one? 😄

**Comment by av-gantimurov** [comment]
> @mandiant/flare-vm should we add bochs to the default flare-vm installation? I think so, I would like to be able to use Bochs in IDA. 🤔 Opinions?

Bochs supported only in **IDA Pro**, IDA Home and IDA Free has only local debugger (+GDBServer in IDA Home).

**Comment by Ana06** [comment]
> Bochs supported only in IDA Pro, IDA Home and IDA Free has only local debugger (+GDBServer in IDA Home).

Good point! I do use IDA Pro 😄 We even have tools in the default configuration that only work with IDA Pro and not outside IDA like the capa plugin (`ida.plugin.capa.vm`) that needs python. I think it is nice that is there after I install IDA Pro in a VM with FLARE-VM. 😉
